MarsEnv.java

```diff
@@ -58,7 +58,18 @@ public boolean executeAction(String ag, Structure action) {
                 model.randMove();
             } else if (action.equals(poop)) {
                 model.maybePoop();
-
+			} else if (action.getFunctor().equals("moveLeft")) {
+				int id = (int)((NumberTerm)action.getTerm(0)).solve(); //add agent id to action command
+				model.moveLeft(id);
+			} else if (action.getFunctor().equals("moveRight")) {
+				int id = (int)((NumberTerm)action.getTerm(0)).solve(); //add agent id to action command
+				model.moveRight(id);
+			} else if (action.getFunctor().equals("moveUp")) {
+				int id = (int)((NumberTerm)action.getTerm(0)).solve(); //add agent id to action command
+				model.moveUp(id);
+			} else if (action.getFunctor().equals("moveDown")) {
+				int id = (int)((NumberTerm)action.getTerm(0)).solve(); //add agent id to action command
+				model.moveDown(id);
             } else {
```

Comment:
Strictly speaking, these aren't "left", "right", etc (that would have to be relative to the agent, no?), but "West", "East", etc.

Reply:
I've resolved this in a downstream branch. Will be addressed in an upcoming PR.